lib/utils.js

```diff
@@ -109,28 +117,37 @@ function requestWithRetry(request, options, callback) {
  * the details of working with auth in the client code.
  *
  * @param {Array<string>} scopes list of scopes to request as part of auth
+ * @param {Object} config an object with extra configuration parameters (such
+ *     as keyFile or key)
  * @return {function(Object, function(=?,=?,=?):?)} request style function
  *     accepting (options, callback)
  */
-function authorizedRequestFactory(scopes) {
-
+function authorizedRequestFactory(scopes, config) {
+  var authClient;
```

Comment:
I want to get thoughts on this - both whether it's ok to have an authClient object "cached" in the closure of this function, and also the behavior that should happen in the following scenario:

1. Agent calls `authorizedRequestFactory`, which returns a function that can asynchronously make an authorized request (we'll call this function `request`).
2. `request` is called for the first time, which leads to a call to `getAuthClient`.
3. Before `authClient` is assigned in the callback that was passed to `getAuthClient` in the previous call, `request` is called again. As a result, `getAuthClient` is called again as well. This means that `authClient` will be assigned multiple times in its callback.

It seems like any additional auth client instances will be garbage collected, but could this be an issue? (On a side note - I believe this issue also existed before, but in a different form.)

If it is, one solution I was thinking was that if step 3 ever occurs, then we could create an event listener that waits until `authClient` is assigned for the first time.